kubectl-volsync/cmd/migration_create.go

```diff
@@ -36,70 +91,319 @@ var migrationCreateCmd = &cobra.Command{
 		}
+		mc.Capacity = &capacity
 	}
-	// The PVC name must be specified, and it needs to be in the right format
-	pvcname, err := cmd.Flags().GetString("pvcname")
+
+	accessMode, err := cmd.Flags().GetString("accessmodes")
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to fetch access mode, %w", err)
 	}
-	if _, err := ParseXClusterName(pvcname); err != nil {
-		return err
+
+	if corev1.PersistentVolumeAccessMode(accessMode) != corev1.ReadWriteOnce &&
+		corev1.PersistentVolumeAccessMode(accessMode) != corev1.ReadOnlyMany &&
+		corev1.PersistentVolumeAccessMode(accessMode) != corev1.ReadWriteMany &&
+		corev1.PersistentVolumeAccessMode(accessMode) != corev1.ReadWriteOncePod {
+		return fmt.Errorf("unsupported access mode: %v", accessMode)
+	}
+	accessModes := []corev1.PersistentVolumeAccessMode{corev1.PersistentVolumeAccessMode(accessMode)}
+	mc.AccessModes = accessModes
+
+	storageClass, err := cmd.Flags().GetString("storageclass")
+	if err != nil {
+		return fmt.Errorf("failed to fetch storageClass, %w", err)
+	}
+	if storageClass == "" {
+		mc.StorageClassName = nil
+	} else {
+		mc.StorageClassName = &storageClass
+	}
+	// For migration use case, only Copymethod="Direct" is supported
+	mc.CopyMethod = volsyncv1alpha1.CopyMethodDirect
+
+	serviceType, err := cmd.Flags().GetString("servicetype")
+	if err != nil {
+		return fmt.Errorf("please provide service type, err = %w", err)
+	}
+
+	if corev1.ServiceType(serviceType) != corev1.ServiceTypeClusterIP &&
+		corev1.ServiceType(serviceType) != corev1.ServiceTypeLoadBalancer {
+		return fmt.Errorf("unsupported service type: %v", corev1.ServiceType(serviceType))
 	}
+	mc.ServiceType = (*corev1.ServiceType)(&serviceType)
+	mc.RDName = mc.Namespace + "-" + mc.PVCName + "-migration-dest"
+
 	return nil
 }
 
-func doMigrationCreate(cmd *cobra.Command, args []string) error {
-	// Create the empty relationship
-	configDir, err := cmd.Flags().GetString("config-dir")
+//nolint:funlen
+func (mc *migrationCreate) newMigrationRelationshipDestination() (
+	*migrationRelationshipDestination, error) {
+	mrd := &migrationRelationshipDestination{}
+
+	// Assign the values from migrationCreate built after parsing cmd args
+	mrd.RDName = mc.RDName
+	mrd.PVCName = mc.PVCName
+	mrd.Namespace = mc.Namespace
+	mrd.Cluster = mc.Cluster
+
+	if mc.PVC == nil {
+		if mc.Capacity == nil {
+			return nil, fmt.Errorf("capacity arg must be provided")
+		}
+		mrd.Destination.Capacity = mc.Capacity
+
+		// Default "AccessMode = ReadWriteOnce" if not provided
+		mrd.Destination.AccessModes = mc.AccessModes
+
+		if mc.StorageClassName == nil {
+			klog.Infof("Storage class is not specified as part of cmd arg, binding to default storage class")
+			mrd.Destination.StorageClassName = nil
+		} else {
+			mrd.Destination.StorageClassName = mc.StorageClassName
+		}
+
+		mrd.Destination.DestinationPVC = &mc.PVCName
+	} else {
+		mrd.Destination.DestinationPVC = &mc.PVC.Name
```

Comment:
Since the operator is never creating the PVC, only mrd.Destination.DestinationPVC needs to be set. The other fields are only used if you need to create the PVC (and should be handled in ensureDestPVC).

Reply:
okay. Keeping only DestinationPVC and service type assignment and removing irrelevant assignments.

Reply:
Once the PVC exists, no I don't think we need the values any more.

My objection isn't really about saving vs not saving them. It's about where they are being saved. They are being put in a `volsyncv1alpha1.ReplicationDestinationRsyncSpec` which implies to the reader (though I realize the code doesn't) that they will be used when creating the ReplicationDestination object.

I'll leave it up to you about whether to save the values... just not in that struct because it makes the code hard to understand.

Reply:
> Once the PVC exists, no I don't think we need the values any more.


Makes sense